### PR TITLE
Basic URL toJSON() test

### PR DESCRIPTION
**Pull request by annevk**
Standard discussion: https://github.com/whatwg/url/issues/137.

### CODE REVIEW

File: url/url-tojson.html [hunk 1]
```diff
@@ -0,0 +1,12 @@
+<!doctype html>
+<meta charset=utf-8>
+<title>URL's toJSON()</title>
+<script src=/resources/testharness.js></script>
+<script src=/resources/testharnessreport.js></script>
+<div id=log></div>
+<script>
+test(() => {
+  const a = new URL("https://example.com/")
+  assert_equals(JSON.stringify(a), a.href)
```
Comment (zcorpan) [inline url/url-tojson.html]: I think this should be `'"' + a.href + '"'`

Comment (annevk) [inline url/url-tojson.html]: Thanks, added a commit.

**Comment by zcorpan** [comment]
Lgtm

File: url/url-tojson.html [hunk 2]
```diff
@@ -0,0 +1,12 @@
+<!doctype html>
+<meta charset=utf-8>
+<title>URL's toJSON()</title>
+<script src=/resources/testharness.js></script>
+<script src=/resources/testharnessreport.js></script>
+<div id=log></div>
+<script>
+test(() => {
+  const a = new URL("https://example.com/")
+  assert_equals(JSON.stringify(a), "\"" + a.href + "\"")
```
Comment (rniwa) [inline url/url-tojson.html]: I usually prefer having the full string in the expectation so that there is no ambiguity.

**Comment by annevk** [comment]
I addressed @rniwa's feedback as well.

**Comment by w3c-bots** [comment]
# Chrome (unstable channel) #
Testing web-platform-tests at revision 92db1a3da4b071306f14a35321d29a8a572b7263
Using browser at version 58.0.3000.4 dev
Starting 10 test iterations
All results were stable
## All results ##
<details>
<summary><a href="http://w3c-test.org/submissions/4702/url/url-tojson.html">/url/url-tojson.html</a></summary>


|     Subtest      | Results |
|------------------|---------|
|                  | OK      |
| `URL's toJSON()` | FAIL    |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/4702/url/interfaces.html">/url/interfaces.html</a></summary>


|                                                   Subtest                                                    | Results |
|--------------------------------------------------------------------------------------------------------------|---------|
|                                                                                                              | OK      |
| `URL interface: attribute search`                                                                            | PASS    |
| `URL interface: new URL("http://foo") must inherit property "password" with the proper type (4)`             | PASS    |
| `URL interface: new URL("http://foo") must inherit property "searchParams" with the proper type (10)`        | PASS    |
| `URL interface: operation toJSON()`                                                                          | FAIL    |
| `URL interface: attribute password`                                                                          | PASS    |
| `URL interface: existence and properties of interface prototype object`                                      | FAIL    |
| `URL interface: new URL("http://foo") must inherit property "host" with the proper type (5)`                 | PASS    |
| `URL interface: attribute origin`                                                                            | PASS    |
| `URLSearchParams interface: operation has(USVString)`                                                        | PASS    |
| `URL interface: new URL("http://foo") must inherit property "port" with the proper type (7)`                 | PASS    |
| `URLSearchParams interface: operation delete(USVString)`                                                     | PASS    |
| `Testing Symbol.iterator property of iterable interface URLSearchParams`                                     | FAIL    |
| `URLSearchParams interface: existence and properties of interface object`                                    | PASS    |
| `Testing pair iterable interface URLSearchParams`                                                            | FAIL    |
| `URLSearchParams interface object name`                                                                      | PASS    |
| `URL interface: new URL("http://foo") must inherit property "username" with the proper type (3)`             | PASS    |
| `URLSearchParams interface: operation append(USVString,USVString)`                                           | PASS    |
| `URLSearchParams interface: operation values()`                                                              | PASS    |
| `URL interface: attribute port`                                                                              | PASS    |
| `URL interface: new URL("http://foo") must inherit property "pathname" with the proper type (8)`             | PASS    |
| `URLSearchParams interface: operation entries()`                                                             | PASS    |
| `URL interface: attribute username`                                                                          | PASS    |
| `URL interface: existence and properties of interface prototype object's "constructor" property`             | PASS    |
| `URL interface: new URL("http://foo") must inherit property "protocol" with the proper type (2)`             | PASS    |
| `URL interface: attribute hash`                                                                              | PASS    |
| `URLSearchParams interface: operation keys()`                                                                | PASS    |
| `URL interface: new URL("http://foo") must inherit property "toJSON" with the proper type (12)`              | FAIL    |
| `URL interface: new URL("http://foo") must inherit property "hash" with the proper type (11)`                | PASS    |
| `URL interface: existence and properties of interface object`                                                | PASS    |
| `URL interface: new URL("http://foo") must inherit property "origin" with the proper type (1)`               | PASS    |
| `URL interface: new URL("http://foo") must inherit property "href" with the proper type (0)`                 | PASS    |
| `URLSearchParams interface object length`                                                                    | PASS    |
| `URLSearchParams interface: operation forEach(function,any)`                                                 | PASS    |
| `URL interface: stringifier`                                                                                 | PASS    |
| `URL interface: attribute href`                                                                              | PASS    |
| `URL interface: attribute searchParams`                                                                      | PASS    |
| `URL interface: attribute protocol`                                                                          | PASS    |
| `URLSearchParams interface: operation set(USVString,USVString)`                                              | PASS    |
| `URLSearchParams interface: existence and properties of interface prototype object`                          | FAIL    |
| `URLSearchParams interface: operation getAll(USVString)`                                                     | PASS    |
| `URL must be primary interface of new URL("http://foo")`                                                     | PASS    |
| `Stringification of new URL("http://foo")`                                                                   | PASS    |
| `URL interface: attribute host`                                                                              | PASS    |
| `URL interface: new URL("http://foo") must inherit property "search" with the proper type (9)`               | PASS    |
| `URL interface: new URL("http://foo") must inherit property "hostname" with the proper type (6)`             | PASS    |
| `URL interface: attribute hostname`                                                                          | PASS    |
| `URLSearchParams interface: stringifier`                                                                     | PASS    |
| `URL interface object length`                                                                                | PASS    |
| `URLSearchParams interface: existence and properties of interface prototype object's "constructor" property` | PASS    |
| `URL interface: attribute pathname`                                                                          | PASS    |
| `URLSearchParams interface: operation get(USVString)`                                                        | PASS    |
| `URL interface object name`                                                                                  | PASS    |

</details>

**Comment by w3c-bots** [comment]
# Firefox (nightly channel) #
Testing web-platform-tests at revision 92db1a3da4b071306f14a35321d29a8a572b7263
Using browser at version BuildID 20170123125947; SourceStamp 36486fdc3813ef7943ae5b07b4128866d1938a6c
Starting 10 test iterations
All results were stable
## All results ##
<details>
<summary><a href="http://w3c-test.org/submissions/4702/url/url-tojson.html">/url/url-tojson.html</a></summary>


|     Subtest      | Results |
|------------------|---------|
|                  | OK      |
| `URL's toJSON()` | FAIL    |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/4702/url/interfaces.html">/url/interfaces.html</a></summary>


|                                                   Subtest                                                    | Results |
|--------------------------------------------------------------------------------------------------------------|---------|
|                                                                                                              | OK      |
| `URL interface: attribute search`                                                                            | PASS    |
| `URL interface: new URL("http://foo") must inherit property "password" with the proper type (4)`             | PASS    |
| `URL interface: new URL("http://foo") must inherit property "searchParams" with the proper type (10)`        | PASS    |
| `URL interface: operation toJSON()`                                                                          | FAIL    |
| `URL interface: attribute password`                                                                          | PASS    |
| `URL interface: existence and properties of interface prototype object`                                      | PASS    |
| `URL interface: new URL("http://foo") must inherit property "host" with the proper type (5)`                 | PASS    |
| `URL interface: attribute origin`                                                                            | PASS    |
| `URLSearchParams interface: operation has(USVString)`                                                        | PASS    |
| `URL interface: new URL("http://foo") must inherit property "port" with the proper type (7)`                 | PASS    |
| `URLSearchParams interface: operation delete(USVString)`                                                     | PASS    |
| `Testing Symbol.iterator property of iterable interface URLSearchParams`                                     | FAIL    |
| `URLSearchParams interface: existence and properties of interface object`                                    | PASS    |
| `Testing pair iterable interface URLSearchParams`                                                            | PASS    |
| `URLSearchParams interface object name`                                                                      | PASS    |
| `URL interface: new URL("http://foo") must inherit property "username" with the proper type (3)`             | PASS    |
| `URLSearchParams interface: operation append(USVString,USVString)`                                           | PASS    |
| `URLSearchParams interface: operation values()`                                                              | PASS    |
| `URL interface: attribute port`                                                                              | PASS    |
| `URL interface: new URL("http://foo") must inherit property "pathname" with the proper type (8)`             | PASS    |
| `URLSearchParams interface: operation entries()`                                                             | PASS    |
| `URL interface: attribute username`                                                                          | PASS    |
| `URL interface: existence and properties of interface prototype object's "constructor" property`             | PASS    |
| `URL interface: new URL("http://foo") must inherit property "protocol" with the proper type (2)`             | PASS    |
| `URL interface: attribute hash`                                                                              | PASS    |
| `URLSearchParams interface: operation keys()`                                                                | PASS    |
| `URL interface: new URL("http://foo") must inherit property "toJSON" with the proper type (12)`              | FAIL    |
| `URL interface: new URL("http://foo") must inherit property "hash" with the proper type (11)`                | PASS    |
| `URL interface: existence and properties of interface object`                                                | PASS    |
| `URL interface: new URL("http://foo") must inherit property "origin" with the proper type (1)`               | PASS    |
| `URL interface: new URL("http://foo") must inherit property "href" with the proper type (0)`                 | PASS    |
| `URLSearchParams interface object length`                                                                    | PASS    |
| `URLSearchParams interface: operation forEach(function,any)`                                                 | PASS    |
| `URL interface: stringifier`                                                                                 | PASS    |
| `URL interface: attribute href`                                                                              | PASS    |
| `URL interface: attribute searchParams`                                                                      | PASS    |
| `URL interface: attribute protocol`                                                                          | PASS    |
| `URLSearchParams interface: operation set(USVString,USVString)`                                              | PASS    |
| `URLSearchParams interface: existence and properties of interface prototype object`                          | PASS    |
| `URLSearchParams interface: operation getAll(USVString)`                                                     | PASS    |
| `URL must be primary interface of new URL("http://foo")`                                                     | PASS    |
| `Stringification of new URL("http://foo")`                                                                   | PASS    |
| `URL interface: attribute host`                                                                              | PASS    |
| `URL interface: new URL("http://foo") must inherit property "search" with the proper type (9)`               | PASS    |
| `URL interface: new URL("http://foo") must inherit property "hostname" with the proper type (6)`             | PASS    |
| `URL interface: attribute hostname`                                                                          | PASS    |
| `URLSearchParams interface: stringifier`                                                                     | PASS    |
| `URL interface object length`                                                                                | PASS    |
| `URLSearchParams interface: existence and properties of interface prototype object's "constructor" property` | PASS    |
| `URL interface: attribute pathname`                                                                          | PASS    |
| `URLSearchParams interface: operation get(USVString)`                                                        | PASS    |
| `URL interface object name`                                                                                  | PASS    |

</details>

**Comment by zcorpan** [comment]
Lgtm, lets merge together with whatwg/url PR

**Pull request merged**